### ISSUE TITLE
test_private_function

### ISSUE THREAD

**Issue by tamaro-skaljic** [rust-lang/rust-clippy#14343]
### What it does

As mentioned in https://doc.rust-lang.org/book/ch11-03-test-organization.html#testing-private-functions, "There’s debate within the testing community about whether or not private functions should be tested directly". Implementing this lint would allow projects to specify wether or not private functions should have tests or not.

### Example

Testing private functions is prohibited in this project.

```rust
fn add(left: usize, right: usize) -> usize {
    left + right
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn add() {
        assert_eq!(add(2, 2), 4);
    }
}
```
